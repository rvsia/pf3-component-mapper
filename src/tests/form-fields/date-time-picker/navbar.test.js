import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Navbar from '../../../form-fields/date-time-picker/navbar';

describe('<Navbar />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      month: new Date('December 17, 2018 00:00:00'),
    };
  });

  it.skip('should render correctly', () => {
    const wrapper = mount(<Navbar { ...initialProps } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it.skip('should render year navbar variant correctly', () => {
    const wrapper = mount(<Navbar { ...initialProps } isYear />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call previsouse click prop', () => {
    const onPreviousClick = jest.fn();
    const wrapper = mount(<Navbar { ...initialProps } onPreviousClick={ onPreviousClick } />);
    wrapper.find('td').first().simulate('click');
    expect(onPreviousClick).toHaveBeenCalled();
  });

  it('should call next click prop', () => {
    const onNextClick = jest.fn();
    const wrapper = mount(<Navbar { ...initialProps } onNextClick={ onNextClick } />);
    wrapper.find('td').last().simulate('click');
    expect(onNextClick).toHaveBeenCalled();
  });

  it('should call onMonthClick click prop', () => {
    const onMonthClick = jest.fn();
    const wrapper = mount(<Navbar { ...initialProps } onMonthClick={ onMonthClick } />);
    wrapper.find('button.navbar-center-button').simulate('click');
    expect(onMonthClick).toHaveBeenCalledWith(true);
  });

  it('should call toggleSelectingYear click prop', () => {
    const toggleSelectingYear = jest.fn();
    const wrapper = mount(<Navbar { ...initialProps } isYear toggleSelectingYear={ toggleSelectingYear } />);
    wrapper.find('button.navbar-center-button').simulate('click');
    expect(toggleSelectingYear).toHaveBeenCalledWith(true);
  });

});
