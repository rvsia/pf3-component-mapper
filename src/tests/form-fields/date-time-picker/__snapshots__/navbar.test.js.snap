// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render correctly 1`] = `
<Navbar
  month={2018-12-16T23:00:00.000Z}
>
  <table
    className="year-interval-header"
  >
    <tbody>
      <tr>
        <td
          onClick={[Function]}
        >
          <Icon
            name="angle-left"
            type="fa"
          >
            <FontAwesome
              name="angle-left"
            >
              <span
                aria-hidden={true}
                className="fa fa-angle-left"
              />
            </FontAwesome>
          </Icon>
        </td>
        <td>
          <button
            className="navbar-center-button"
            onClick={[Function]}
            type="button"
          >
            December 2018
          </button>
        </td>
        <td
          onClick={[Function]}
        >
          <Icon
            name="angle-right"
            type="fa"
          >
            <FontAwesome
              name="angle-right"
            >
              <span
                aria-hidden={true}
                className="fa fa-angle-right"
              />
            </FontAwesome>
          </Icon>
        </td>
      </tr>
    </tbody>
  </table>
</Navbar>
`;

exports[`<Navbar /> should render year navbar variant correctly 1`] = `
<Navbar
  isYear={true}
  month={2018-12-16T23:00:00.000Z}
>
  <table
    className="year-interval-header"
  >
    <tbody>
      <tr>
        <td
          onClick={[Function]}
        >
          <Icon
            name="angle-left"
            type="fa"
          >
            <FontAwesome
              name="angle-left"
            >
              <span
                aria-hidden={true}
                className="fa fa-angle-left"
              />
            </FontAwesome>
          </Icon>
        </td>
        <td>
          <button
            className="navbar-center-button"
            onClick={[Function]}
            type="button"
          >
            2018
          </button>
        </td>
        <td
          onClick={[Function]}
        >
          <Icon
            name="angle-right"
            type="fa"
          >
            <FontAwesome
              name="angle-right"
            >
              <span
                aria-hidden={true}
                className="fa fa-angle-right"
              />
            </FontAwesome>
          </Icon>
        </td>
      </tr>
    </tbody>
  </table>
</Navbar>
`;
